core/packages/test/scripts/fetch-beacon-data.sh

```diff
@@ -9,14 +9,26 @@ echo "finalized_block_number is: $finalized_block_number"
 
 # get block_root by block_number
 finalized_block_root=$(curl -s "$beacon_endpoint_http/eth/v1/beacon/blocks/$finalized_block_number/root" | jq -r ".data.root")
-echo "finalized_block_root is: $finalized_block_root"
+echo "finalized_block_root is:" $finalized_block_root
 
 # get beacon header by block_root
 beacon_header=$(curl -s "$beacon_endpoint_http/eth/v1/beacon/headers/$finalized_block_root")
-echo $beacon_header
+echo "beacon header is:" $beacon_header
 
-# get beacon block by block_root
+# get validators root from genesis
+validatorsRoot=$(curl -s "$beacon_endpoint_http/eth/v1/beacon/genesis" | jq -r ".data.genesis_validators_root")
+echo "validators root is:" $validatorsRoot
+
+# get checkpoint
+checkpoint_root=$(curl -s "$beacon_endpoint_http/eth/v1/beacon/states/head/finality_checkpoints" | jq -r ".data.finalized.root")
+echo "checkpoint_root is:" $checkpoint_root
+curl -s "$beacon_endpoint_http/eth/v1/beacon/light_client/bootstrap/$checkpoint_root" | jq -r "." > testdata/beacon_check_point.json
+
+# get beacon block
+echo "fetch beacon block..."
 curl -s "$beacon_endpoint_http/eth/v2/beacon/blocks/$finalized_block_root" | jq -r "." > testdata/beacon_block.json
 
-# get beacon state by block_number
-curl -s "$beacon_endpoint_http/eth/v2/debug/beacon/states/$finalized_block_number" | jq -r "." > testdata/beacon_state.json
+if [[ $beacon_endpoint_http == "http://127.0.0.1:9596" ]];then
```

Comment:
Why do we get the beacon state for local testnet only? 🤔

Reply:
Since we've provided [script](https://github.com/Snowfork/snowbridge/blob/ron/sno-405/core/packages/test/scripts/start-goerli.sh) to setup goerli nodes from locally I would suggest we fetch beacon state of goerli also in this way.
Btw, I’ve never successfully get beacon state from public [endpoint](https://lodestar-goerli.chainsafe.io) before, :joy:  guess related to the bad network in China.